test/datasets/containers.jl

```diff
@@ -0,0 +1,53 @@
+include("imports.jl")
+
+@testset ExtendedTestSet "TableDataset" begin
+
+    @testset ExtendedTestSet "TableDataset from rowaccess table" begin
+        Tables.columnaccess(::Type{<:Tables.MatrixTable}) = false
+        Tables.rowaccess(::Type{<:Tables.MatrixTable}) = true
+        
+        testtable = Tables.table([1 4.0 "7"; 2 5.0 "8"; 3 6.0 "9"])
+        td = TableDataset(testtable)
+
+        @test getobs(td, 1) == [1, 4.0, "7"]
+        @test nobs(td) == 3
+    end
+
+    @testset ExtendedTestSet "TableDataset from columnaccess table" begin
+        Tables.columnaccess(::Type{<:Tables.MatrixTable}) = true
+        Tables.rowaccess(::Type{<:Tables.MatrixTable}) = false
+        
+        testtable = Tables.table([1 4.0 "7"; 2 5.0 "8"; 3 6.0 "9"])
+        td = TableDataset(testtable)
+
+        @test getobs(td, 2) == [2, 5.0, "8"]
+        @test nobs(td) == 3
+    end
+
+    @testset ExtendedTestSet "TableDataset from DataFrames" begin
+        testtable = DataFrame(
+            col1=[1, 2, 3, 4, 5], 
+            col2=["a", "b", "c", "d", "e"], 
+            col3=[10, 20, 30, 40, 50], 
+            col4=["A", "B", "C", "D", "E"],
+            col5=[100., 200., 300., 400., 500.],
+            split=["train", "train", "train", "valid", "valid"]
+        )
+        @inferred TableDataset TableDataset(testtable)
```

Comment:
What are you trying to test here? Maybe this should be
```julia
@test TableDataset(testtable) isa TableDataset{<:DataFrame}
```

Reply:
Yeah this makes more sense. I have updated the tests accordingly.